src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -0,0 +1,201 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using Microsoft.Spark.Interop.Ipc;
+using Microsoft.Spark.Interop.Internal.Java.Util;
+using Microsoft.Spark.Interop;
+using System.Collections.Generic;
+
+namespace Microsoft.Spark
+{
+    /// <summary>
+    /// A broadcast variable. Broadcast variables allow the programmer to keep a read-only variable
+    /// cached on each machine rather than shipping a copy of it with tasks. They can be used, for
+    /// example, to give every node a copy of a large input dataset in an efficient manner. Spark
+    /// also attempts to distribute broadcast variables using efficient broadcast algorithms to
+    /// reduce communication cost.
+    /// </summary>
+    [Serializable]
+    public sealed class Broadcast<T>: IJvmObjectReferenceProvider
+    {
+        [NonSerialized]
+        private readonly JvmObjectReference _jvmObject;
+        private readonly long _bid;
+
+        internal Broadcast(SparkContext sc, object value, JvmObjectReference sparkContext)
+        {
+            var path = (string)Path.Combine(sc.TempDir, Path.GetRandomFileName());
+            Version version = SparkEnvironment.SparkVersion;
+
+            // For Spark versions 2.3.0 and 2.3.1, Broadcast variable is created through different
+            // functions.
+            if (version.Major == 2 && version.Minor == 3 && (version.Build == 0 || 
```

Comment:
Can we create a function for each case? Also, you may want to add a check for unsupported version. Refer to `CommandProcessor.ReadSqlCommands`

Reply:
Made the change, please let me know if this looks fine. Thanks!